src/components/functionselect/functionselect.js

```diff
@@ -14,25 +14,49 @@ angular.module('vlui')
 
         scope.func = {
           selected: undefined,
-          list: [undefined]
+          list: {
+            aboveFold: [],
+            belowFold: [] // could be empty
+          }
         };
 
-        function getFns(type) {
+        // timeUnits for T
+        var timeUnits = {
+          aboveFold: [undefined, "year", "quarter", "month", "date", "day", "hours", "minutes", "seconds", "yearmonthdate"],
+          belowFold: ["hoursminutes", "hoursminutesseconds", "milliseconds", 
+            "minutesseconds", "secondsmilliseconds", "yearmonth", "yearmonthdatehours", 
+            "yearmonthdatehoursminutes", "yearmonthdatehoursminutesseconds", "yearquarter"]
+        }
+
+        // aggregations for Q
+        var aggregations = {
+          aboveFold: [undefined, "min", "max", "median", "mean", "sum"],
+          belowFold: ["average", "distinct",
+            "missing", "modeskew", "q1", "q3", "stdev", "stdevp", "valid", "values", "variance", "variancep"] // there is no "count" for Q
+        }
+
+        function getTimeUnits(type) {
           if (type === 'temporal') {
-            return Schema.schema.definitions.TimeUnit.enum;
+            if (!timeUnits.all || timeUnits.all.length <= 0) {
+              timeUnits.all = timeUnits.aboveFold.concat(timeUnits.belowFold);
+            }
+            return timeUnits.all;
           }
           return [];
         }
 
-        function getAggrs(type) {
+        function getAggregations(type) {
           if(!type) {
             return [COUNT];
           }
 
           // HACK
           // TODO: make this correct for temporal as well
           if (type === 'quantitative' ){
-            return Schema.schema.definitions.AggregateOp.enum;
+            if (!aggregations.all || aggregations.all.length <= 0) {
```

Comment:
I don't understand why you have to do lazy initialization for `aggregations.all`  why don't you just initialize it right after when you define `aggregations`?

Plus, please rename this to `aggregates` for consistency.